src/meta/src/manager/catalog/mod.rs

```diff
@@ -164,51 +168,135 @@ where
         }
     }
 
-    pub async fn drop_database(&self, database_id: DatabaseId) -> MetaResult<NotificationVersion> {
+    pub async fn drop_database(
```

Comment:
Better to add some comments explaining what's the procedure, including why `Vec<CatalogDeletedId>` is returned.